src/scripts/install_machine.py

```diff
@@ -0,0 +1,362 @@
+#! /usr/bin/python3
+
+import traceback
+import json
+import subprocess
+import sys
+import os
+import xml.etree.ElementTree as ET
+import tempfile
+import logging
+
+
+def virsh(connectionName, args):
+    cmd = ["virsh", "-c", f"qemu:///{connectionName}"]
+    for arg in args:
+        cmd.append(arg)
+
+    logging.debug("Running virsh command :" + ' '.join(cmd))
+
+    return subprocess.run(cmd, check=False, capture_output=True)
+
+
+def assertVmExists(args):
+    cmd = f"virsh -c qemu:///{args.get('connectionName')} list --all | awk  " + "'{print $2}'" + f" | grep -q --line-regexp --fixed-strings {args.get('vmName')}"
+
+    subprocess.run(cmd, shell=True, check=True, capture_output=True)
```

Comment:
Probably needs more than a simple substring check, but you get the idea.

```suggestion
def assert_vm_exists(connection, name):
    assert name in virsh(connection, 'list', '--all')
```